### ISSUE TITLE
Kernel fails tests/language/case_expression_with_assignment_test.dart

### ISSUE THREAD

**Issue by rmacnak-google** [dart-lang/sdk#29379]
```
const ERROR_A = 0;
const ERROR_B = 1;

errorToString(error) {
  switch (error) {
    case ERROR_A:
      return "ERROR_A";
    case ERROR_B = 1:         //# 01: compile-time error
    case ERROR_B:             //# none: ok
      return "ERROR_B";
    default:
      return "Unknown error";
  }
}
```

**Comment by peter-ahe-google** [comment]
Thank you. I have fixed this in my local branch.

**Comment by peter-ahe-google** [comment]
Proposed fix in [CL 2911023002](https://codereview.chromium.org/2911023002).
